### PR TITLE
have more config and move all the config into the JRubyJarTask

### PR DESCRIPTION
**Pull request by mkristian**

### CODE REVIEW

File: jruby-gradle-jar-plugin/src/main/groovy/com/github/jrubygradle/jar/JRubyJarTask.groovy [hunk 1]
```diff
@@ -16,7 +22,198 @@ import org.gradle.api.tasks.StopExecutionException
...
+        // there are other duplicates as well :(
+        setDuplicatesStrategy(DuplicatesStrategy.EXCLUDE)
+
+        project.gradle.taskGraph.addTaskExecutionListener(
+            new TaskExecutionListener() {
+                void afterExecute(Task task, TaskState state) {
+                }
+
+                void beforeExecute(Task task) {
+                      println task
```
Comment (rtyler) [inline jruby-gradle-jar-plugin/src/main/groovy/com/github/jrubygradle/jar/JRubyJarTask.groovy]: These println statements should be nuked

**Comment by rtyler** [comment]
One minor thing but overall this looks reasonable to me from a standpoint. Either this change or a subsequent pull request should update the README to omit the use of the `jruby{}` configuration closure.

I want @ysb33r to :+1:  before we merge.

File: jruby-gradle-jar-plugin/src/main/groovy/com/github/jrubygradle/jar/JRubyJarTask.groovy [hunk 2]
```diff
@@ -16,7 +22,196 @@ import org.gradle.api.tasks.StopExecutionException
...
+
+    void jrubyVersion(String version) {
+      this.jrubyVersion = version
+    }
+
+    void jrubyMainsVersion(String version) {
+      this.jrubyMainsVersion = version
+    }
+
+    void jruby(Closure cfg) {
```
Comment (ysb33r) [inline jruby-gradle-jar-plugin/src/main/groovy/com/github/jrubygradle/jar/JRubyJarTask.groovy]: Nice. Might add deprecation annotation as well as ading the `deprecated` word somewhere in the log message.

File: jruby-gradle-jar-plugin/src/main/groovy/com/github/jrubygradle/jar/JRubyJarTask.groovy [hunk 3]
```diff
@@ -16,7 +22,196 @@ import org.gradle.api.tasks.StopExecutionException
...
+        project.logger.info 'It is no longer necessary to use the jruby closure on a JRubyJar task.' 
+        def cl = cfg.clone()
+        cl.delegate = this
+        cl.call()
+    }
+  
+    void applyConfig() {
```
Comment (ysb33r) [inline jruby-gradle-jar-plugin/src/main/groovy/com/github/jrubygradle/jar/JRubyJarTask.groovy]: Should make this private I reckon.

**Comment by ysb33r** [comment]
I good with it overall. Just the two comments I've added should be dealt with, then it should be good to merge.

**Comment by rtyler** [comment]
Adding a note that this may fix #137

**Comment by mkristian** [comment]
let's move on with #140 - in case I missed something here or we feel the need to improve something we can do so with #140

**Pull request merged**